### DISCUSSION TITLE
Update prompt-toolkit-3.0.40 ?

### DISCUSSION THREAD

**Discussion by arthursw** [xonsh/xonsh#5434] [Q&A] [answered]
`prompt-toolkit-3.0.40` conflicts with `ipython` which requires `prompt-toolkit<3.1.0,>=3.0.41`. 

I get the following error when installing `xonsh`:
```
[...]
Using cached prompt_toolkit-3.0.40-py3-none-any.whl (385 kB)
Installing collected packages: prompt-toolkit
  Attempting uninstall: prompt-toolkit
    Found existing installation: prompt-toolkit 3.0.43
    Uninstalling prompt-toolkit-3.0.43:
      Successfully uninstalled prompt-toolkit-3.0.43
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
ipython 8.24.0 requires prompt-toolkit<3.1.0,>=3.0.41, but you have prompt-toolkit 3.0.40 which is incompatible.
Successfully installed prompt-toolkit-3.0.40
```

I had reinstalled `ipython` after `xonsh` because it was broken.

Is it possible to update `prompt-toolkit-3.0.40`?

**Comment by anki-code** [comment]
1. What if you will use [xonsh-install](https://github.com/anki-code/xonsh-install) to install xonsh in isolated env?
2. To unpin prompt-toolkit we need more tests and probably fixes. 
3. If you can install the latest version of ptk manually (i.e. `xpip install prompt-toolkit=3.0.43`) and live with this and report about stability it will be cool.

**Reply by arthursw** [reply]
I guess it would work, but it would be nice to update prompt-toolkit anyway?

**Reply by arthursw** [reply]
Ok, I'll tell you if I have any problem with `prompt-toolkit=3.0.43`

**Reply by anki-code** [reply]
We pined it becuase of issue found (https://github.com/xonsh/xonsh/pull/5403). But I have no strong confidence that it's really dependent to ptk. We need more tests.

**Reply by arthursw** [reply]
Ok, thanks!

**Comment by anki-code** [comment]
Prompt-toolkit will be unpinned in the future release (https://github.com/xonsh/xonsh/pull/5438).

**Comment by Spitfire1900** [comment]
@arthursw  - use [xonsh-install](https://github.com/anki-code/xonsh-install) or pipx

**Reply by anki-code** [reply]
hey @Spitfire1900 please give a star to xonsh-install if you like it. It helps to make it more visible for xonsh users.

**Comment by anki-code** [comment, marked as answer]
@arthursw ptk unpinned in 0.17.0 release - https://github.com/xonsh/xonsh/releases/tag/0.17.0
